consensus/hotstuff/signature/randombeacon_inspector.go

```diff
@@ -0,0 +1,117 @@
+		}
+		return fmt.Errorf("unexpected error verifying beacon signature from %d: %w", signerIndex, err)
+	}
+
+	if !verif { // invalid signature
+		return fmt.Errorf("invalid beacon share from %d: %w", signerIndex, signature.ErrInvalidFormat)
+	}
+	return nil
+}
+
+// TrustedAdd adds a share to the internal signature shares store.
+// There is no pre-check of the signature's validity _before_ adding it.
+// It is the caller's responsibility to make sure the signature was previously verified.
+// Nevertheless, the implementation guarantees safety (only correct threshold signatures
+// are returned) through a post-check (verifying the threshold signature
+// _after_ reconstruction before returning it).
+// The function is thread-safe but locks its internal state, thereby permitting only
+// one routine at a time to add a signature.
+// Returns:
+//  - (true, nil) if the signature has been added, and enough shares have been collected.
+//  - (false, nil) if the signature has been added, but not enough shares were collected.
+//  - (false, error) if there is any exception adding the signature share.
+//      - engine.InvalidInputError if signerIndex is invalid (out of the valid range)
+//  	- engine.DuplicatedEntryError if the signer has been already added
+//      - other error if there is an unexpected exception.
+func (r *randomBeaconInspector) TrustedAdd(signerIndex int, share crypto.Signature) (enoughshares bool, exception error) {
+	// Trusted add to the crypto layer
+	enough, err := r.follower.TrustedAdd(signerIndex, share)
+	if err != nil {
+		if crypto.IsInvalidInputsError(err) {
+			return false, engine.NewInvalidInputErrorf("trusted add from %d failed: %w", signerIndex, err)
+		}
+		if crypto.IsDuplicatedSignerError(err) {
+			return false, engine.NewDuplicatedEntryErrorf("trusted add from %d failed: %w", signerIndex, err)
+		}
+		return false, fmt.Errorf("unexpected error while adding share from %d: %w", signerIndex, err)
+	}
+	return enough, nil
+}
+
+// EnoughShares indicates whether enough shares have been accumulated in order to reconstruct
+// a group signature.
+//
+// The function is write-blocking
+func (r *randomBeaconInspector) EnoughShares() bool {
+	return r.follower.EnoughShares()
+}
+
+// Reconstruct reconstructs the group signature. The function is thread-safe but locks
+// its internal state, thereby permitting only one routine at a time.
+//
+// Returns:
+// - (signature, nil) if no error occured
+// - (nil, crypto.notEnoughSharesError) if not enough shares were collected
+// - (nil, crypto.invalidInputsError) if at least one collected share does not serialize to a valid BLS signature,
+//    or if the constructed signature failed to verify against the group public key and stored message. This post-verification
+//    is required  for safety, as `TrustedAdd` allows adding invalid signatures.
+// - (nil, error) for any other unexpected error.
+// The function is blocking.
```

Comment:
I think this
> The function is blocking.

is redundant with the [statement above](https://github.com/onflow/flow-go/blob/96765647eae016b87ab641ac09614c958e876c36/consensus/hotstuff/signature/randombeacon_inspector.go#L104-L105)
> The function is thread-safe but locks its internal state, thereby permitting only one routine at a time.

```suggestion
```